CONTRIBUTING.md

```diff
@@ -0,0 +1,85 @@
+# Contributing to OSRD
+
+All contributions, bug reports, bug fixes, documentation improvements, enhancements, and ideas are welcome.
+
+Check out our community's [Code of Conduct](https://github.com/DGEXSolutions/osrd/blob/dev/CODE_OF_CONDUCT.md) and feel free to say hi on [libera chat](https://web.libera.chat/#osrd) if you'd like. It's a nice place to chat about OSRD development, ask questions, and get to know the other contributors and users in a less formal setting. If you are not sure about what to do have a look at currently open [good first issues](https://github.com/DGEXSolutions/osrd/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22).
+
+## Getting oriented
+
+The OSRD project has 2 primary repos:
+
+1. [`osrd`](https://github.com/DGEXSolutions/osrd): This is where the simulator is developed.
+2. [`osrd-website`](https://github.com/DGEXSolutions/osrd-website): Where the [`official website`](https://osrd.fr), blog, documentation, ... are hosted.
+
+The `osrd` repo itself contains many smaller packages.
+
+Some packages of interest (more details in packages' `README.md`):
+
+- [`core`](https://github.com/DGEXSolutions/osrd/tree/dev/core): The core engine of the simulation where physics, pathfinding and conflict detection are computed.
+- [`editoast`](https://github.com/DGEXSolutions/osrd/tree/dev/editoast): Service to retrieve data from the databases and make data available through http endpoints.
+  It is also the place where edition of the infrastructure is computed.
+- [`front`](https://github.com/DGEXSolutions/osrd/tree/dev/front): GUI of OSRD.
+- [`tests`](https://github.com/DGEXSolutions/osrd/tree/dev/tests): Package to run integration tests on services.
+- [`python/osrd_schemas`](https://github.com/DGEXSolutions/osrd/tree/dev/python/osrd_schemas): Definition of the railjson format.
+
+## What we are trying to build
+
+OSRD is an open source web application for railway infrastructure design, capacity analysis, timetabling and simulation.
+Check out details on the [website specific page](https://osrd.fr/en/about/).
+
+OSRD also currently has the following "development process" goals:
+
+- **Don't merge everything ... don't merge too early**: Every feature we add increases maintenance burden and compile times.
+  Only merge features that are "generally" useful.
+  Don't merge major changes or new features unless we have relative consensus that the design is correct and that we have the developer capacity to support it.
+- **Control and consistency over 3rd party code reuse**: Only add a dependency if it is absolutely necessary.
+  Every dependency we add decreases our autonomy and consistency.
+- **Don't re-invent every wheel**: As a counter to the previous point, don't re-invent everything at all costs.
+  If there is a dependency in the ecosystem that is the "de-facto" standard, we should heavily consider using it.
```

Comment:
I don't quite get the point of mentioning this here. It might make sense for bevy, does it make sense for us?

Reply:
I think it is the case yes. For instance `rocket` was kind of such situation where we were using a dependency that was no longer the defacto-standard and we had to migrate to actix.
Dependency management can be a mess and having a lot of them could lead to security issues so we have to be careful when adding one.

Reply:
I think indeed that it would be better placed in the General Rules section